compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3749,6 +3783,28 @@ private boolean isAssignableToUnionType(BType source, BType target, Set<TypePair
         return true;
     }
 
+    private boolean containsDistinctError(Set<BType> errorTypesSet) {
+        List<BType> errorTypes = new ArrayList<>(errorTypesSet);
+        int size = errorTypes.size();
+        Set<BTypeIdSet.BTypeId> t1TypeIds = ((BErrorType) errorTypes.get(0)).typeIdSet.getAll();
+        for (int i = 1; i < size; i++) {
+            Set<BTypeIdSet.BTypeId> t2TypeIds = ((BErrorType) errorTypes.get(i)).typeIdSet.getAll();
+            if (!t1TypeIds.equals(t2TypeIds)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
for empty type ids also this will be true; isn't it?

```
type Foo error;
type Bar error;

Foo|Bar e = error Foo("");
```

Reply:
Changed